File: Supervised Learning/Multi-class Classification/One-Vs-All Logistic Regression/Octave/predictImg.m
function p = predictImg(Theta, Img)
    %imfinfo(Img)
    X = imread(Img); % reads the image (20x20)

    X = double(X);% converts it to double
    temp = X;% creates a copy for later use

    X = (X.-128)./255;%normalize the features
    X = X .* (temp > 0);%return the original 0 values to the X
    X = reshape(X, [], numel(X));%converts the 20x20 matrix into a 1x400 vector

    displayData(X);%display the image imported

    p = predictOneVsAll(Theta, X)

    if p==10
        p=0;
    end

    fprintf('\nImage is number: %d\n', p);

end

File: Supervised Learning/Multi-class Classification/One-Vs-All Logistic Regression/Python/predictImg.py
import loadData as data
import numpy as np
import predictOneVsAll as pred
import scipy.misc

import oneVsAll as onevAll


def predictImg(Theta, Img):

    X = scipy.misc.imread(Img)

    X = np.double(X)  # converts it to double
    temp = np.copy(X)   # creates a copy for later use

    X = (X-128)/255  # normalize the features
    X = X * (temp > 0) #return the original 0 values to the X
    X = X.T
    X = np.reshape(X, X.size)
    X = np.array(X)[np.newaxis]


    p = pred.calculatePredictOneVsAll(Theta, X).item()

    if p==10:
        p=0

    return p



def main():

    lmda = 0.1
    num_labels = 10
    theta = onevAll.oneVsAll(num_labels, lmda)


    image = './predict-images/4.png'

    p = predictImg(theta, image)
    print("Image is number: %d" % p)



# If this script is executed, then main() will be executed
if __name__ == '__main__':
    main()
